### ISSUE TITLE
Is the python sandboxed?

### ISSUE THREAD

**Issue by NoelJacob** [flet-dev/serious-python#126]
Can we enable the users to run untrusted python code off the internet and not have access to filesystem? For all platforms that is.

**Comment by ndonkoHenri** [comment]
No. Python runs **in-process**, inside your Flutter app — `serious_python_run` is resolved out of the app's own process through the [dart_bridge](https://github.com/flet-dev/dart-bridge) FFI bridge (3.0.0 replaced the old socket transport), so the interpreter has exactly the permissions your app has, on every platform.

serious_python adds no Python-level sandbox (and there are probably no plans to add one) — CPython has no supported way to isolate untrusted code inside a host process. The only boundary is the OS app sandbox you enable yourself (`com.apple.security.app-sandbox` on macOS — note our own example apps ship with it set to `false`; on iOS/Android the platform sandbox applies to the whole app, not to Python specifically). Running untrusted third-party code isn't something this package is designed for.
